public/controlpanel.php

```diff
@@ -6,13 +6,7 @@
 require_once __DIR__ . '/../vendor/autoload.php';
 require_once __DIR__ . '/../lib/bootstrap.php';
 
-if (RA_ReadCookieCredentials($user, $points, $truePoints, $unreadMessageCount, $permissions)) {
-    if (getAccountDetails($user, $userDetails) == false) {
-        // Immediate redirect if we cannot validate user!
-        header("Location: " . getenv('APP_URL') . "?e=accountissue");
-        exit;
-    }
-} else {
+if (!RA_ValidateCookie($user, $permissions, $userDetails)) {
```

Comment:
`RA_ValidateCookie` returns the `$userDetails` without requiring a second call to fetch them.

For the second call to fail after the first succeeded, the account would have to still have a valid cookie, but be marked for deletion. This logic is now included in `RA_ValidateCookie`.